### PR TITLE
[Feature](runtime filter) normalize ignore runtime filter

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

normalize ignore runtime filter
图片

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run  buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/exprs/minmax_predicate.h [hunk 1]
```diff
@@ -48,104 +46,71 @@ class MinMaxNumFunc : public MinMaxFuncBase {
...
-    bool find(void* data) override {
-        if (data == nullptr) {
-            return false;
-        }
-
-        T val_data = *reinterpret_cast<T*>(data);
-        if constexpr (NeedMin) {
-            if (val_data < _min) {
-                return false;
+    void update_batch(const vectorized::ColumnPtr& column, const vectorized::NullMap& nullmap,
```
Comment (HappenLee) [inline be/src/exprs/minmax_predicate.h:91]: nullmap seems useless？

File: be/src/exprs/runtime_filter.h [hunk 2]
```diff
@@ -135,15 +135,19 @@ struct RuntimeFilterParams {
 struct FilterFuncBase {
 public:
     void set_filter_id(int filter_id) {
+        _is_runtime_filter = true;
```
Comment (HappenLee) [inline be/src/exprs/runtime_filter.h:138]: we really need the bool ？ check `(_filter_id == -1)` do the same work ?

File: be/src/olap/accept_null_predicate.h [hunk 3]
```diff
@@ -208,6 +165,49 @@ class AcceptNullPredicate : public ColumnPredicate {
     }
 
 private:
+    uint16_t _evaluate_inner(const vectorized::IColumn& column, uint16_t* sel,
+                             uint16_t size) const override {
+        if (column.has_null()) {
```
Comment (HappenLee) [inline be/src/olap/accept_null_predicate.h:170]: cast to ColumnNullable to reduce the virtual fun call?

File: be/src/olap/accept_null_predicate.h [hunk 4]
```diff
@@ -208,6 +165,49 @@ class AcceptNullPredicate : public ColumnPredicate {
...
+            // create selected_flags
+            uint16_t max_idx = *std::max_element(sel, sel + size);
+            auto selected_flags_ptr = std::make_unique<bool[]>(max_idx + 1);
```
Comment (HappenLee) [inline be/src/olap/accept_null_predicate.h:173]: the sel size lower than 4096? better stack mem?

File: be/src/olap/comparison_predicate.h [hunk 5]
```diff
@@ -362,7 +365,25 @@ class ComparisonPredicateBase : public ColumnPredicate {
         _evaluate_vec_internal<true>(column, size, flags);
     }
```
Comment (HappenLee) [inline be/src/olap/comparison_predicate.h:367]: Add a comment for the value

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**